packages/web-components/fast-foundation/src/combobox/combobox.ts

```diff
@@ -608,26 +586,46 @@ export class FASTCombobox extends FormAssociatedCombobox {
     /**
      * Calculate and apply listbox positioning based on available viewport space.
      *
-     * @param force - direction to force the listbox to display
      * @public
      */
     public setPositioning(): void {
-        const currentBox = this.getBoundingClientRect();
-        const viewportHeight = window.innerHeight;
-        const availableBottom = viewportHeight - currentBox.bottom;
-
-        this.position = this.forcedPosition
-            ? this.positionAttribute
-            : currentBox.top > availableBottom
-            ? SelectPosition.above
-            : SelectPosition.below;
-
-        this.positionAttribute = this.forcedPosition
-            ? this.positionAttribute
-            : this.position;
-
-        this.maxHeight =
-            this.position === SelectPosition.above ? ~~currentBox.top : ~~availableBottom;
+        if (this.$fastController.isConnected) {
+            this.cleanup = autoUpdate(this, this.listbox, async () => {
+                const { middlewareData, x, y } = await computePosition(
+                    this,
+                    this.listbox,
+                    {
+                        placement: "bottom",
+                        strategy: "fixed",
+                        middleware: [
+                            flip(),
```

Comment:
Am I correct in assuming that though is is positioned to the bottom, if there's not enough space, this will "flip" it to the top?

Reply:
Yep, that's what this middleware does: https://floating-ui.com/docs/flip